### ISSUE TITLE
Automate update for github-action-config.json

### ISSUE THREAD

**Issue by sayboras** [golangci/golangci-lint#1208]
Need to investigate on how to automate this

_Originally posted by @sayboras in https://github.com/golangci/golangci-lint/pull/1207#issuecomment-653829660_

Currently we have Make target `make assets/github-action-config.json` to update github-action-config.json, will be better to have it run as part of release automatically.

Updated: Just found the below steps in https://github.com/golangci/golangci-lint/blob/master/docs/src/docs/contributing/workflow.mdx

After making a release you need to update (TODO: automate partially):

1. GitHub [action config](https://github.com/golangci/golangci-lint/blob/master/assets/github-action-config.json) by running:

```sh
make assets/github-action-config.json
```

2. Update netlify state hash

```sh
make update_netlify_state
```

3. Contributors list

```sh
make update_contributors_list # may take 15 min
```
`

**Comment by SVilgelm** [comment]
So, actually we need to run these steps after a release. It means, that when the release workflow is completed, next one should run those steps, create a commit an push to the master branch

**Comment by sayboras** [comment]
> So, actually we need to run these steps after a release. It means, that when the release workflow is completed, next one should run those steps, create a commit an push to the master branch

Yes, you are right. Currently, I think we just wait till the issue with gorelease for bew gets resolved first, then the above steps can be done in the same job (after gorelease)

**Comment by denis-tingajkin** [comment]
@sayboras I feel to commit changes automatically we need a user account/bot. 

Question: Have `golangci` a user for automatization?

**Comment by ernado** [comment]
AFAIK changes can be commited via default github actions user, but I can be wrong

**Comment by sayboras** [comment]
@denis-tingajkin @ernado I also think github action user can do so, similar to goreleaser is able to create the release. Might need to test it out. I did have dependabot commit directly into master in my repo before.

**Comment by SVilgelm** [comment]
@sayboras it's easy, except on part. The master branch is protected, we need to create the Pull Request from an action.
1. the `release` job is completed
2. git checkout -b update-docs
3. `make update_contributors_list`, `make assets/github-action-config.json`, `make update_netlify_state`
4. git add -u
5. git push origin update-docs
6. magic script to create a pull request using GitHub API

**Comment by SVilgelm** [comment]
I know that the default GITHUB_TOKEN does not allow creation of the pull requests, but we can use `secrets.GOLANGCI_LINT_TOKEN`

**Comment by SVilgelm** [comment]
or if the `secrets.GOLANGCI_LINT_TOKEN` has admin permissions, then we can push to master branch directly
